templates/shared/pack.js

```diff
@@ -0,0 +1,12 @@
+const { readFileSync } = require('fs');
+const { parse, resolve } = require('path');
+const AdmZip = require('adm-zip');
+
+const { base } = parse(__dirname);
+const { version } = JSON.parse(
+  readFileSync(resolve(__dirname, 'build', 'manifest.json'), 'utf8')
+);
+
+const zip = new AdmZip();
+zip.addLocalFolder('build');
+zip.writeZip(`${base}-v${version}.zip`);
```

Comment:
Can you wrap this code block inside a try-catch to handle errors?

Reply:
I have a question, a simple try-catch is necessary? For example, under the current context,

```js
try {
  // Do packing or throw an error
} catch (e) {
  console.error(e);
}
```

is equivalent to:

```js
// Do packing or throw an error
```

Are you expecting more?

Reply:
`try-catch` is necessary because with it we can add a custom success and error message.

Reply:
Okay, but is `console.error(e)` what you think sufficient, or something else?

Reply:
Yes `console.error()` would be sufficient and we can log a custom error message instead of showing `adm-zip`'s error message.
Something like this - `Error! Failed to generate a zip file.`.

Reply:
We can use this one - `Error! Failed to generate a zip file.`.